src/main/java/org/jabref/gui/importer/fetcher/WebSearchPaneViewModel.java

```diff
@@ -130,16 +144,26 @@ public void search() {
         }
 
         SearchBasedFetcher activeFetcher = getSelectedFetcher();
+        Callable<ParserResult> parserResultCallable = () -> new ParserResult(activeFetcher.performSearch(query));
+        String fetcherName = activeFetcher.getName();
+
+        Optional<DOI> doi = DOI.findInText(query);
+        if (doi.isPresent()) {
+            DoiFetcher doiFetcher = new DoiFetcher(preferencesService.getImportFormatPreferences());
+            parserResultCallable = () -> new ParserResult(OptionalUtil.toList(doiFetcher.performSearchById(doi.get().getDOI())));
+            fetcherName = doiFetcher.getName();
+        }
+
+        String finalFetcherName = fetcherName;
```

Comment:
Then I would also state that in the code:

```suggestion
        final String finalFetcherName = fetcherName;
```

Reply:
Got it, I've updated the PR with these changes